rust/tool/src/install.rs

```diff
@@ -28,11 +28,13 @@ pub struct Installer {
     configuration_limit: usize,
     esp_paths: EspPaths,
     generation_links: Vec<PathBuf>,
+    target_arch: Architecture,
```

Comment:
```suggestion
    architecture: Architecture,
```

Let's drop the `target_`. Cross compiling (terminology) is hard enough. As far as it concerns the installer, host, build, and target platform are all the same.

Reply:
Maybe I got something wrong, but installer can install on a build != host != target platform.
Here, host is the UEFI platform of the target.

Reply:
(except if we have lzbt touching efivars which is another problem.)

Reply:
I don't think the installer can or should install from one architecture to another (e.g. from aarch64 -> x86). The only case where this would happen is when remote installing (of which I'm not really convinced).

That the host platform for the binaries is UEFI (and that these binaries are usually cross-compiled from x86 or aarch64) doesn't really matter for the installer. It only matters for the derivation that builds the stub.

Reply:
While all of this might be true today, the installer is run as part of the switch-to-configuration phase, going forward if the day where you can "pre-switch-to-configuration" nixos systems, every install bootloader phase will have to deal with potentially build != host != target platforms.

I'm okay with assuming for now that the installer cannot deal with this scenario for now, but I don't believe it's a far-fetched thing and it's more correct.

> That the host platform for the binaries is UEFI (and that these binaries are usually cross-compiled from x86 or aarch64) doesn't really matter for the installer. It only matters for the derivation that builds the stub.

Yes, of course, it's just a correctness thing so we avoid being confused in the future if someone come up with unexpected usecases IMHO.